File: coffee-system-api/src/main/java/com/jesperancinha/coffee/system/utils/ExecutorServiceHelper.java
package com.jesperancinha.coffee.system.utils;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.TimeUnit;

/**
 * Created by joaofilipesabinoesperancinha on 05-05-16.
 */
public class ExecutorServiceHelper {
    public static void shutDownExecutorService(ExecutorService executorService) {
        if(executorService!= null)
        {
            executorService.shutdown();
            try {
                executorService.awaitTermination(60, TimeUnit.SECONDS);
            } catch (InterruptedException e) {
                executorService.shutdownNow();
            }
        }
    }
}

File: coffee-system/src/main/kotlin/org/jesperancinha/coffee/system/api/utils/ExecutorServiceHelper.kt
package org.jesperancinha.coffee.system.api.utils

import java.util.concurrent.ExecutorService
import java.util.concurrent.TimeUnit

/**
 * Created by joaofilipesabinoesperancinha on 05-05-16.
 */
object ExecutorServiceHelper {
    fun shutDownExecutorService(executorService: ExecutorService) {
        executorService.shutdown()
        try {
            executorService.awaitTermination(60, TimeUnit.SECONDS)
        } catch (e: InterruptedException) {
            executorService.shutdownNow()
        }
    }
}